src/opensearch.py

```diff
@@ -70,6 +72,50 @@ def install(self):
             logger.error(f"Failed to install/upgrade opensearch. \n{e}")
             raise OpenSearchInstallError()
 
+    @override
+    def is_service_started(self, paused: Optional[bool] = False) -> bool:
+        """Check if the snap service and JVM process are running.
+
+        Set paused=True if the process was intentionally paused.
+        """
+        if not self._opensearch.present:
+            return False
+
+        if not service_running("snap.opensearch.daemon.service"):
+            return False
+
+        # Now, we must dig deeper into the actual status of systemd and the JVM process.
+        # First, we want to make sure the process is not stopped, dead or zombie.
+        # From: https://github.com/torvalds/linux/blob/master/fs/proc/array.c#L126-L140
+        # "Parked" state is ignored as it applies to threads.
+        try:
+            pid = subprocess.check_output(["lsof", "-ti:9200"], text=True).rstrip()
+            if not pid or not os.path.exists(f"/proc/{pid}/stat"):
+                return False
+            with open(f"/proc/{pid}/stat") as f:
+                stat = f.read()
+        except subprocess.CalledProcessError:
+            return False
+
+        if stat[2] == "T" and paused:
```

Comment:
could you add a comment above this line and explain what `stat[2] == "T"` refers to?